src/Api/Authorization/HeaderHandler.cs

```diff
@@ -89,7 +91,28 @@ protected override async Task HandleForbiddenAsync(AuthenticationProperties prop
                 Status = StatusCodes.Status403Forbidden,
                 Type = "https://docs.passwordless.dev/guide/errors.html#forbidden",
             },
-        });
+        };
+
+        var endpoint = Context.GetEndpoint();
+        if (endpoint == null)
+        {
+            await _problemDetailsService.WriteAsync(context);
+            return;
+        }
```

Comment:
I would suggest un-inverting this statement (i.e. remove `return` and instead do `if (endpoint != null)`), so that `await _problemDetailsService.WriteAsync(context);` only appears once.